pallets/pips/src/lib.rs

```diff
@@ -1174,4 +1215,16 @@ impl<T: Trait> Module<T> {
         <ProposalVotes<T>>::insert(id, proposer, vote);
         Ok(())
     }
+
+    /// Returns a reportable representation of a proposal taking care that the reported data are not
+    /// too large.
+    fn reportable_proposal_data(proposal: T::Proposal) -> ProposalData {
```

Comment:
Proposal could be huge, so we could use here as a reference, and `encode` is going to make another sort of copy of it.